src/frontend/apps/desk/src/features/auth/useAuthStore.tsx

```diff
@@ -1,40 +1,47 @@
 import { create } from 'zustand';
 
-import { initKeycloak } from './keycloak';
+import { UserData, getMe } from '@/features/auth/api';
 
 interface AuthStore {
   authenticated: boolean;
   initAuth: () => void;
   initialized: boolean;
   logout: () => void;
-  token: string | null;
+  userData?: UserData;
 }
 
 const initialState = {
   authenticated: false,
   initialized: false,
-  token: null,
+  userData: undefined,
 };
 
 export const useAuthStore = create<AuthStore>((set) => ({
   authenticated: initialState.authenticated,
   initialized: initialState.initialized,
-  token: initialState.token,
+  userData: initialState.userData,
 
   initAuth: () =>
     set((state) => {
-      if (process.env.NEXT_PUBLIC_KEYCLOAK_LOGIN && !state.initialized) {
-        initKeycloak((token) => set({ authenticated: true, token }));
-        return { initialized: true };
+      if (state.initialized) {
+        return {};
       }
-
-      /**
-       * TODO: Implement OIDC production authentication
-       */
-
+      getMe()
+        .then((data: UserData) => {
+          set({ authenticated: true, userData: data });
+          return { initialized: true };
+        })
+        .catch(() => {
+          // todo - implement a proper login screen to prevent automatic navigation.
+          window.location.replace(
+            new URL('authenticate', process.env.NEXT_PUBLIC_API_URL).href,
+          );
+        });
       return {};
     }),
```

Comment:
About this conversation (https://github.com/numerique-gouv/people/pull/59#discussion_r1492177860)

It will fix the duplicate call, getMe is async:
```suggestion
      getMe()
        .then((data: UserData) => {
          set({ authenticated: true, userData: data });
        })
        .catch(() => {
          // todo - implement a proper login screen to prevent automatic navigation.
          window.location.replace(
            new URL('authenticate', process.env.NEXT_PUBLIC_API_URL).href,
          );
        });
      return { initialized: true };
```

Reply:
I will apply your suggestion !

Reply:
Actually, this suggestion is not working. It triggers dozens of `getMe` request, while mine initial code was only triggering request twice due to the React strict mode.